### PR TITLE
Remove DefaultConvert functions

### PR DESCRIPTION
**Pull request by wojtek-t**
This is no longer used anywhere.

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/conversion/converter.go [hunk 1]
```diff
@@ -523,14 +501,6 @@ func (c *Converter) convert(sv, dv reflect.Value, scope *scope) error {
 		return c.callUntyped(sv, dv, f, scope)
 	}
 
-	return c.defaultConvert(sv, dv, scope)
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/conversion/converter.go]: are you planning to remove the reflect-based conversion in a follow-up?

Comment (wojtek-t) [reply staging/src/k8s.io/apimachinery/pkg/conversion/converter.go]: I would like to, but there are still places where this is actually used (I didn't see that used in apiserver in our e2e tests, but I've seen that in both integration and unit tests still).
Unless I missed anything, the conversions that are still happening and doesn't have auto-generated code that "self-conversions" (I mean conversion of the object to exactly the same type and version). 
So the questions to answer are:
1) whether the tests are too artificial and should be fixed to not do that
2) or whether it really can happen in practise too, and maybe we should auto-generate self-conversions too

My understanding is that it should be (1), but I didn't have enough time to convince myself.
E.g. one of the places to analyze that is causing problems in some tests is this one:
https://github.com/kubernetes/kubernetes/pull/84441/files#diff-6e3d2200e9c4c629141574d861645ddfR32

File: staging/src/k8s.io/apimachinery/pkg/conversion/converter.go [hunk 2]
```diff
@@ -138,10 +138,6 @@ type Scope interface {
 	// parameters, you'll run out of stack space before anything useful happens.
 	Convert(src, dest interface{}, flags FieldMatchingFlags) error
 
-	// DefaultConvert performs the default conversion, without calling a conversion func
-	// on the current stack frame. This makes it safe to call from a conversion func.
-	DefaultConvert(src, dest interface{}, flags FieldMatchingFlags) error
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/conversion/converter.go]: cc @smarterclayton for use downstream? I know you still want to drop reflect-based conversions and require everything to be generated... this seems like a decent first step

Comment (smarterclayton) [reply staging/src/k8s.io/apimachinery/pkg/conversion/converter.go]: I know there are places we use it but have a registered conversion.  I don’t know of a reason those places can’t just call it directly, and anyone else using direct conversion as a “copy but drop fields” is easily fixed to “copy explicit”.

Kill it.

**Pull request merged**